src/toil/batchSystems/kubernetes.py

```diff
@@ -52,8 +53,25 @@
 from toil.lib.humanize import human2bytes
 from toil.resource import Resource
 
+from toil.lib.retry import retry
+
 logger = logging.getLogger(__name__)
+     
+def retryable_kubernetes_errors(e):
+    """
+    A function that determins whether or not Toil should retry or stop given 
+    exceptions thrown by Kubernetes. 
+    """
+    if isinstance(e, urllib3.exceptions.MaxRetryError) or e.response.status_code == 504 or isinstance(e, ApiException):
+        return False
+    return True
```

Comment:
The sense is still backward here, unfortunately.

This version of the check is better, but what if `e` doesn't have a `response`, or its `response` is some type that doesn't have a `status_code`? You should do something like `getattr(getattr(e, 'response', None), 'status_code', None) == 504` so it's true if the value is present and also equal to 504.

Reply:
Right thank you ....